spec/components/panel/tag_component_spec.rb

```diff
@@ -0,0 +1,16 @@
+require 'spec_helper'
+
+RSpec.describe Tag::TagComponent, type: :component do
+  it 'should return the correct HTML' do
+    component = Tag::TagComponent.new(text: 'Alert', colour: 'green')
+
+    html = render_inline(component).to_html
+
+    expect(html).to eql(<<~html
+      <strong class="govuk-tag govuk-tag--green">
+        Alert
+      </strong>
+      html
+    )
```

Comment:
This is different from the existing tests. I wonder if this makes it easier to check whether or not the HTML is correctly formed. For me this was easier to do then translating it into `have_css` assertions like the previous tests.

Reply:
This is similar to [Jest inline snapshots](https://jestjs.io/docs/en/snapshot-testing) from the JavaScript world, can confirm it is a very efficient and sane way for testing small blocks of HTML or JSON.

One difference is that `jest` (`rspec` equivalent) gives you a button (`jest --updateSnapshot`) to update all out of date snapshots, which makes refactoring tests en-masse very easy for banal changes.

Reply:
I tend to lean towards the `have_css` approach because it's ignorant of whitespace, indentation etc.

```ruby
expect(html).to have_css('strong', class: 'govuk-tag govuk-tag--green', text: 'Alert')
```
On smaller, simpler components this is fine and readable so is fine by me

Reply:
Yeah agree re: whitespace @peteryates. That's where having a tool do it for you comes in really handy.

(This exists but it doesn't look maintained https://github.com/yesmeck/rspec-snapshot)

Reply:
🤔 If we render the components on a page we could use jest to snapshot the component. We could go one step further and do some visual regression testing using jest-image-snapshot (developed by american express) https://www.npmjs.com/package/jest-image-snapshot